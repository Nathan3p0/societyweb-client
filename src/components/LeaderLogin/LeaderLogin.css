.leaderLogin__form h3 {
    color: #fff;
    margin: 2rem 0 1rem 0;
    font-size: 2rem;
    text-align: center;
}


.leaderLogin__form li {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin-top: .5rem
}

.leaderLogin__form > li > label {
    flex: 1 0 100px;
    color: #fff;
    font-size: 1.2rem;
}

.leaderLogin__form > li > label + * {
    flex: 1 0 220px;
}

.button__container {
    justify-content: flex-end;
}

.leaderLogin__form li button:first-of-type {
    margin-right: .5rem;
}

.leaderLogin__form li button {
    padding: 12px 24px;
    border: none;
    background: #4ab3dd;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

.leaderLogin__form li button:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: 1px 1px 5px #333333;
}

.leaderLogin__form li input {
    padding: 15px;
    border: 1px solid #00000025;
    border-radius: 5px;
}