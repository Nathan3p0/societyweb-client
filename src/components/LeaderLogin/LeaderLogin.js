import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import AuthApiService from '../../services/auth-api-service';
import LoginInfoContext from '../../context/LoginInfoContext';
import './LeaderLogin.css'

class LeaderLogin extends Component {
    static contextType = LoginInfoContext;

    static defaultProps = {
        leaderLoginSuccess: () => { }
    }

    state = {
        error: null,
        redirect: false
    }

    handleLeaderLoginSubmit = (e) => {
        e.preventDefault();
        this.setState({
            error: null
        });

        const { username, password } = e.target;

        AuthApiService.postLogin({
            username: username.value,
            password: password.value
        })
            .then(res => {
                username.value = '';
                password.value = '';
                this.setState({
                    redirect: true
                })
            })
            .catch(res => {
                this.setState({
                    error: res.error
                })
            })
    }

    handleLeaderDemoLogin = () => {
        this.setState({
            error: null
        });

        AuthApiService.postLogin({
            username: 'demoAdmin',
            password: 'Password1.'
        })
            .then(res => {
                this.setState({
                    redirect: true
                })
            })
            .catch(res => {
                this.setState({
                    error: res.error
                })
            })
    }

    render() {
        const { error } = this.state;

        if (this.state.redirect) {
            return <Redirect to="/admin" />
        }

        return (
            <form onSubmit={this.handleLeaderLoginSubmit}>
                <ul className="leaderLogin__form">
                    <h3>Team Leader Login</h3>
                    {error && <p className="error">{error}</p>}
                    <li>
                        <label htmlFor="leader-username">Username:</label>
                        <input type="text" name="username" id="leader-username" placeholder="Username" required />
                    </li>
                    <li>
                        <label htmlFor="leader-password">Password:</label>
                        <input type="password" name="password" id="leader-password" placeholder="Password" required />
                    </li>
                    <li className="button__container">
                        <button type="submit">Login</button>
                        <button type="button" onClick={this.handleLeaderDemoLogin}>Demo</button>
                    </li>
                </ul>
            </form>
        );
    }
}

export default LeaderLogin;