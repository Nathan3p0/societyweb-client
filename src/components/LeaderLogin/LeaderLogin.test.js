import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import LeaderLogin from './LeaderLogin'

describe('Leader Login Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<LeaderLogin />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<LeaderLogin />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})