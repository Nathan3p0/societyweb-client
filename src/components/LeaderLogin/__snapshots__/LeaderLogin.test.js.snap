// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leader Login Component Rendering Tests renders to UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <ul
    className="leaderLogin__form"
  >
    <h3>
      Team Leader Login
    </h3>
    <li>
      <label
        htmlFor="username"
      >
        Username:
      </label>
      <input
        id="username"
        name="username"
        placeholder="username"
        required={true}
        type="text"
      />
    </li>
    <li>
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        required={true}
        type="password"
      />
    </li>
    <li>
      <button
        type="submit"
      >
        Login
      </button>
    </li>
  </ul>
</form>
`;
